src/EmailProtection.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Use this as place to store any state or functionality related to Email Protection
+ */
+class EmailProtection {
+    /** @type {string|null} */
+    #previous = null;
+
+    /** @param {import("./DeviceInterface/InterfacePrototype").default} device */
+    constructor (device) {
+        this.device = device
+    }
+
+    /** @returns {boolean} */
+    get generated () {
```

Comment:
Should this be `isGenerated` or `hasGenerated`? At first glance, I was expecting it to hold the actual value.